service-compiler/src/main/java/io/techery/analytics/compiler/model/AnalyticActionClass.java

```diff
@@ -51,11 +50,10 @@ private void obtainAttributes(Elements elementUtils, TypeElement typeElement) {
       if (superTypeElement != null) {
          obtainAttributes(elementUtils, superTypeElement);
       }
+
       if (containsActionPathParam) {
-         if (actionPartFieldName == null) {
-            if (ActionClassUtils.checkHasAnnotatedField(elementUtils, typeElement, ActionPart.class)) {
-               actionPartFieldName = ActionClassUtils.getAnnotatedFieldName(elementUtils, typeElement, ActionPart.class);
-            }
+         if (ActionClassUtils.checkHasAnnotatedField(elementUtils, typeElement, KeyPath.class)) {
```

Comment:
there should be validation or TODO for validation: if there are more pathes then `KeyPath` fields